crates/iceberg-catalog/migrations/20240620151544_views.sql

```diff
@@ -0,0 +1,97 @@
+create type view_format_version as enum ('v1');
+
+create table view
+(
+    view_id             uuid primary key default uuid_generate_v1mc(),
+    CONSTRAINT "tabular_ident_fk" FOREIGN KEY (view_id) REFERENCES tabular (tabular_id),
+    view_format_version view_format_version not null
+);
+
+call add_time_columns('"view"');
+select trigger_updated_at('"view"');
+
+
+create table view_properties
+(
+    property_id uuid primary key default uuid_generate_v1mc(),
+    view_id     uuid not null REFERENCES view (view_id) ON DELETE CASCADE,
+    key         text not null,
+    value       text not null
+);
+
+
+call add_time_columns('view_properties');
+select trigger_updated_at('"view_properties"');
+
+create table view_schema
+(
+    schema_uuid uuid primary key default uuid_generate_v1mc(),
+    schema_id   int   not null,
+    view_id     uuid  not null REFERENCES view (view_id) ON DELETE CASCADE,
+    -- the schema object is quite complex and I'm not sure about the benefits of inviting that complexity into sql
+    schema      jsonb not null,
+    CONSTRAINT "unique_schema_per_view" unique (view_id, schema_id)
+);
+call add_time_columns('view_schema');
+select trigger_updated_at('view_schema');
+
+create table view_version
+(
+    view_version_uuid    uuid        not null primary key default uuid_generate_v1mc(),
```

Comment:
If we change the name above, we should also change it here - maybe to view_version_unique_id

Reply:
changed